tasks/main.yml

```diff
@@ -48,21 +48,30 @@
   with_fileglob:
     - /etc/udev/rules.d/70-*
 
-- name: remove HWADDR from ifcfg-*
+- name: remove HWADDR from ifcfg-* on RHEL/Centos
   lineinfile:
     dest: "{{ item }}"  # 'path' is new in 2.3
     state: absent
     regexp: "^HWADDR=.*"
   with_fileglob:
     - /etc/sysconfig/network-scripts/ifcfg-*
+  when: ansible_os_family == "RedHat"
 
-- name: remove UUID from ifcfg-*
+- name: remove UUID from ifcfg-* on RHEL/Centos
   lineinfile:
     dest: "{{ item }}"  # 'path' is new in 2.3
     state: absent
     regexp: "^UUID=.*"
   with_fileglob:
     - /etc/sysconfig/network-scripts/ifcfg-*
+  when: ansible_os_family == "RedHat"
+
+- name: remove hwaddr information from network interfaces on Ubuntu/Debian
+  lineinfile:
+    dest: "/etc/network/interfaces"  # 'path' is new in 2.3
+    state: absent
+    regexp: "^[ \t]*hwaddress.*"
```

Comment:
I would use `\s*` this is equivalent to the set `[ \t\n\r\f\v]*` but it is up to you ... not necessary ...